### PR TITLE
ComboBox: Added prop to close callout when user it typing

### PR DESCRIPTION
**Pull request by kellygorr**
#### Pull request checklist

[0] Addresses an existing issue: Fixes #0000
[X] Include a change request file using `$ npm run change`

#### Description of changes
Once the callout is opened and the user clicks the input to start typing, we need the callout to close.

Before editing this code I originally tried setting onRenderContainer to null when the input is clicked. This does close the callout like needed, but the isOpen state becomes out of sync.  That is why I created the inputWithoutCallout prop and separated the input click event from the chevron (IconButton) click event.

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts [hunk 1]
```diff
@@ -107,6 +107,12 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
    * Add additional content below the callout list.
    */
   onRenderLowerContent?: IRenderFunction<IComboBoxProps>;
+
+  /**
+   * Close callout when user is typing in input (relies on AllowFreeform: true)
+   * @default false;
+   */
+  inputWithoutCallout?: boolean;
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: You should accomplish this by adding a dismissMenu function on the comboBox to close the callout. That will allow you to get the behavior you are looking for and other authors who may want to dismiss the other menu for another reason will be able to do that as well.

**Comment by dzearing** [comment]
tslint build error:

/home/travis/build/OfficeDev/office-ui-fabric-react/packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx
ERROR: 1310:7  one-line  misplaced 'else'

File: packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts [hunk 2]
```diff
@@ -30,6 +30,11 @@ export interface IComboBoxProps extends ISelectableDroppableTextProps<IComboBox>
   componentRef?: (component: IComboBox) => void;
 
   /**
+   * If there is a menu open this will dismiss the menu
+   */
+  dismissMenu?: () => void;
```
Comment (jspurlin) [inline packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: This is not correct. The function should be exposed via the IComboBox interface and then you can call it from the comboBox componentRef

Comment (kellygorr) [reply packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.Props.ts]: This is a whole lot easier than I originally made it, thank you!

**Pull request merged**